precovery/frame_db.py

```diff
@@ -781,19 +785,22 @@ def new_data_file(self, dataset_id: str, year_month_str: str):
         self.data_files[self._current_data_file_name(dataset_id, year_month_str)] = f
 
     def defragment(self, new_index: FrameIndex, new_db: "FrameDB"):
```

Comment:
The defrag code definitely has been neglected over the last few months. We've been fairly consistent in making our input observations sorted by exposure_id and mjd_utc which minimizes the impact of a defrag, but I think this code is going to become more important for the quirkier datasets.